File: AvaliaProf-app/src/app/shared/models/usuario.ts
export class Usuario{

    public id:number;
    public nome:string;
    public matricula:string;
    public email:string;
    public senha:string;

    constructor(
         id:number,
         nome:string='',
         matricula:string='',
         email:string='',
         senha:string=''
    ){
        this.id = id;
        this.nome = nome;
        this.matricula = matricula;
        this.email = email;
        this.senha = senha;
    }
}

File: AvaliaProf-back/src/main/java/pweb/avaliaprof/modelo/Usuario.java
package pweb.avaliaprof.modelo;


import jakarta.persistence.*;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private long id;
    private String nome;
    private String senha;
    private String email;
    private String matricula;

    public Usuario() {
    }
    public Usuario(String nome, String senha, String email, String matricula) {
        this.nome = nome;
        this.senha = senha;
        this.email = email;
        this.matricula = matricula;
    }

    public String getNome() {
        return this.nome;
    }

    public String getSenha() {
        return this.senha;
    }

    public String getEmail() {
        return this.email;
    }

    public String getMatricula() {
        return this.matricula;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setMatricula(String matricula) {
        this.matricula = matricula;
    }

    public long getId() {
        return this.id;
    }
}
